linera-graphql-client/tests/test.rs

```diff
@@ -0,0 +1,192 @@
+        request,
+        service::{applications::Variables, Applications},
+    };
+
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    let network = Network::Grpc;
+    let mut local_net = LocalNetwork::new(network, 4).unwrap();
+    let client = local_net.make_client(network);
+    local_net.generate_initial_validator_config().await.unwrap();
+    client.create_genesis_config().await.unwrap();
+    local_net.run().await.unwrap();
+
+    let chain_id = client.get_wallet().default_chain().unwrap();
+    let mut node_service = client.run_node_service(None).await.unwrap();
+    let data = request::<Applications, _>(
+        &reqwest::Client::new(),
+        &format!("http://localhost:{}/", node_service.port()),
+        Variables { chain_id },
+    )
+    .await
+    .unwrap()
+    .applications;
+    assert_eq!(data, Vec::new());
+    node_service.assert_is_running();
+}
+
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_blocks_query() {
+    use linera_graphql_client::{
+        request,
+        service::{blocks::Variables, Blocks},
+    };
+
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    let network = Network::Grpc;
+    let mut local_net = LocalNetwork::new(network, 4).unwrap();
+    let client = local_net.make_client(network);
+    local_net.generate_initial_validator_config().await.unwrap();
+    client.create_genesis_config().await.unwrap();
+    local_net.run().await.unwrap();
+
+    let chain_id = client.get_wallet().default_chain().unwrap();
+    let mut node_service = client.run_node_service(None).await.unwrap();
+    let data = request::<Blocks, _>(
+        &reqwest::Client::new(),
+        &format!("http://localhost:{}/", node_service.port()),
+        Variables {
+            chain_id,
+            from: None,
+            limit: None,
+        },
+    )
+    .await
+    .unwrap()
+    .blocks;
+    assert_eq!(data, Vec::new());
+    node_service.assert_is_running();
+}
+
+#[test_log::test(tokio::test)]
+async fn test_end_to_end_block_query() {
+    use linera_graphql_client::{
+        request,
+        service::{block::Variables, Block},
+    };
+
+    let _guard = INTEGRATION_TEST_GUARD.lock().await;
+    let network = Network::Grpc;
+    let mut local_net = LocalNetwork::new(network, 4).unwrap();
+    let client = local_net.make_client(network);
+    local_net.generate_initial_validator_config().await.unwrap();
+    client.create_genesis_config().await.unwrap();
+    local_net.run().await.unwrap();
+
+    let chain_id = client.get_wallet().default_chain().unwrap();
+    let mut node_service = client.run_node_service(None).await.unwrap();
+    let data = request::<Block, _>(
+        &reqwest::Client::new(),
+        &format!("http://localhost:{}/", node_service.port()),
+        Variables {
+            chain_id,
+            hash: None,
+        },
+    )
+    .await
+    .unwrap()
+    .block;
+    assert_eq!(data, None);
+    node_service.assert_is_running();
+}
```

Comment:
We could create a block and an application in some of these tests, so the data isn't empty. (Or maybe even merge them, so the setup creating the block has to be done only once?)

Reply:
yes it is possible.